UnitsNet/QuantityTypeConverter.cs

```diff
@@ -0,0 +1,205 @@
+    }
+
+    /// <summary>
+    /// This attribute defines the unit the quantity has when converting to string
+    /// </summary>
+    public class DisplayAsUnitAttribute : DefaultUnitAttribute
+    {
+        /// <summary>
+        /// The formating used when the quantity is converted to string
+        /// </summary>
+        public string Format { get; set; }
+
+        /// <summary>
+        /// Constructor for DisplayAsUnit attribute
+        /// </summary>
+        /// <param name="unitType">The unit the quantity should be displayed in</param>
+        /// <param name="format">Formating string <see cref="IQuantity.ToString(System.IFormatProvider)"/></param>
+        public DisplayAsUnitAttribute(object unitType, string format = "") : base(unitType) { Format = format; }
+    }
+
+    /// <summary>
+    ///     Converts IQuantitys from string and to string.
+    ///     Implements the TypeConverter interface so that eg the PropertyGrid can read and write the properties implementing the IQuantity interface.
+    /// </summary>
+    /// <typeparam name="TQuantity">Quantity value type, such as <see cref="Length"/> or <see cref="Mass"/>.</typeparam>
+    public class UnitsNetTypeConverter<TQuantity> : TypeConverter where TQuantity : IQuantity
+    {
+        /// <summary>
+        ///     Returns whether this converter can convert an object of the given type to the type of this converter, using the specified context.
+        /// </summary>
+        /// <param name="context">An System.ComponentModel.ITypeDescriptorContext that provides a format context.</param>
+        /// <param name="sourceType">A System.Type that represents the type you want to convert from.</param>
+        /// <returns>true if this converter can perform the conversion; otherwise, false.</returns>
+        public override bool CanConvertFrom(ITypeDescriptorContext context, Type sourceType)
+        {
+            return (sourceType == typeof(string)) || base.CanConvertFrom(context, sourceType);
+        }
+
+        private static TAttribute GetAttribute<TAttribute>(ITypeDescriptorContext context) where TAttribute : UnitAttributeBase
+        {
+            TAttribute attribute = null;
+            AttributeCollection ua = context?.PropertyDescriptor.Attributes;
+
+            attribute = (TAttribute)ua?[typeof(TAttribute)];
+
+            if(attribute != null)
+            {
+                QuantityType expected = default(TQuantity).Type;
+                QuantityType actual = Quantity.From(1, attribute.UnitType).Type;
+                if(expected != actual)
+                {
+                    throw new InvalidOperationException($"The specified UnitType:'{attribute.UnitType}' dose not match QuantityType:'{expected}'");
+                }
+            }
+
+            return attribute;
+        }
+
+        /// <summary>
+        ///     Converts the given object to the type of this converter, using the specified context and culture information.
```

Comment:
If we only support strings, maybe we should say so here instead of being vague about what types it will convert?
Or are there other types too that work?

Same above for CanConvertFrom method.